arcsi/api/item.py

```diff
@@ -277,173 +280,11 @@ def add_item():
-                    no_error = False
-            # this branch is typically used for pre-uploading live episodes (no audio)
-            else: 
-                if not image_file_name:
-                    no_error = False
-
-        # archive files if asked
-        if new_item.archive_lahmastore:
-            if no_error and (play_file or image_file):
-                if image_file_name:
-                    new_item.image_url = archive(
-                        archive_base=new_item.shows[0].archive_lahmastore_base_url,
-                        archive_file_name=image_file_name,
-                        archive_idx=new_item.number,
-                    )
-                    if not new_item.image_url:
-                        no_error = False
-                if new_item.play_file_name:
-                    new_item.archive_lahmastore_canonical_url = archive(
-                        archive_base=new_item.shows[0].archive_lahmastore_base_url,
-                        archive_file_name=new_item.play_file_name,
-                        archive_idx=new_item.number,
-                    )
-
-                    if new_item.archive_lahmastore_canonical_url:
-                        # Only set archived if there is audio data otherwise it's live episode
-                        new_item.archived = True
-                    else:  # Upload didn't succeed
-                        no_error = False
-
-        # broadcast episode if asked
-        if new_item.broadcast and no_error:
-            if not (play_file and image_file):
-                no_error = False
-            else:
-                new_item.airing = broadcast_audio(
-                    archive_base=new_item.shows[0].archive_lahmastore_base_url,
-                    archive_idx=new_item.number,
-                    broadcast_file_name=new_item.play_file_name,
-                    broadcast_playlist=new_item.shows[0].playlist_name,
-                    broadcast_show=new_item.shows[0].name,
-                    broadcast_title=new_item.name,
-                    image_file_name=image_file_name,
-                )
-                if not new_item.airing:
-                    no_error = False
-
-            # TODO some mp3 error
-            # TODO Maybe I used vanilla mp3 not from azuracast
-            # item_audio_obj = MP3(item_path)
-            # return item_audio_obj.filename
-            # item_length = item_audio_obj.info.length
-
-        db.session.commit()
-        # TODO no_error is just bandaid for proper exc handling
-        if no_error:
-            return make_response(
-                jsonify(item_schema.dump(new_item)),
-                200,
-                headers,
-            )
-        else:
-            return "Some error happened, check server logs for details. Note that your media may have been uploaded (to DO and/or Azurcast)."
-
+    return add_item()
 
 @arcsi.route("item/<id>/listen", methods=["GET"])
```

Comment:
@tuz666 isn't `@auth_token_required` missing here?

Reply:
Yes, I forgot about it, nice catch!